src/server/editorServices.ts

```diff
@@ -760,6 +761,7 @@ namespace ts.server {
         readonly toCanonicalFileName: (f: string) => string;
 
         public readonly host: ServerHost;

```

Comment:
I like this idea. In that case, couldn't we get rid of a `fshost` member entirely, since `host` would correctly delegate to `fshost` when needed?

Reply:
We would need fsHost for updateFileSystem call ?

Reply:
Well, I ended up getting rid of fshost except as a parameter when constructing objects, and FileServerHost in the same locations.

I actually think it might make sense to drop FileServerHost in favour of VirtualServerHost in those cases, but I'm not sure whether that type should be used in prominent parts of the API.